src/BloomExe/Workspace/WorkspaceView.cs

```diff
@@ -188,14 +184,44 @@ void AdjustToolPanelLocation(bool allowNarrowing)
 		/// Adjust the tool panel location when the chosen localization changes.
 		/// See https://jira.sil.org/browse/BL-1212 for what can happen if we
 		/// don't adjust.  At the moment, we only widen, we never narrow the
-		/// area allotted to the tab buttons.
+		/// overall area allotted to the tab buttons.  Button widths adjust
+		/// themselves automatically to their Text width.  There doesn't seem
+		/// to be a built-in mechanism to limit the width to a given maximum
+		/// so we implement such an operation ourselves.
 		/// </summary>
 		void HandleTabTextChanged(object sender, EventArgs e)
 		{
+			var btn = sender as Messir.Windows.Forms.TabStripButton;
+			if (btn != null)
+			{
+				const string kEllipsis = "\u2026";
+				// preserve the original string as the tooltip
+				if (!btn.Text.EndsWith(kEllipsis))
+					btn.ToolTipText = btn.Text;
+				// Ensure the button width is no more than 110 pixels by truncating the
+				// Text as needed, with an ellipsis appended to show truncation has occurred.
+				if (btn.Width > 110)
+				{
+					var text = btn.Text;
+					using (Graphics g = btn.Owner.CreateGraphics())
+					{
+						var txtWidth = g.MeasureString(text, btn.Font).Width;
+						var padding = btn.Width - txtWidth;
+						while (txtWidth + padding >= 110)
+						{
+							var len = text.Length - 2;
+							if (len <= 0)
+								break;	// I can't conceive this happening, but I'm also paranoid.
+							text = text.Substring(0, len) + kEllipsis;	// trim, add ellipsis
+							txtWidth = g.MeasureString(text, btn.Font).Width;
+						}
```

Comment:
The biggest problem with a library function is that btn is not a Control or UserControl, which limits its usefulness to ToolStripItem.  But that still might be worth doing I suppose.